custom_openvpn_tunnel.rst

```diff
@@ -0,0 +1,176 @@
+
+For these reasons, caution and attention to detail are strongly advised when performing this procedure.
+
+Configure the VPN
+-----------------
+
+1. Place the configuration file in the correct Directory
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+1. Copy the ``myvpn.ovpn`` file to the directory ``/etc/openvpn/``. Use SCP or a similar tool to transfer the file: ::
+
+    scp myvpn.ovpn root@<NethSecurity_IP>:/etc/openvpn/
+
+
+2. Make sure to set the correct file permissions: ::
+
+    chmod 644 /etc/openvpn/myvpn.ovpn
+    chown root:root /etc/openvpn/myvpn.ovpn
+
+3. Create a new OpenVPN Client configuration in UCI
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+1. Add a new OpenVPN section in the UCI database called ``myvpn``, link the configuration file to this section and enable the VPN ::
+
+    uci add openvpn openvpn
+    uci rename openvpn.@openvpn[-1]='myvpn'
+    uci set openvpn.myvpn.enabled='1'
+    uci set openvpn.myvpn.config='/etc/openvpn/myvpn.ovpn'
+
+2. Commit the changes to save the configuration: ::
+
+    uci commit openvpn
+   
+4. Start the VPN client immediately
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+To start the VPN client without rebooting the system, run: ::
+
+    /etc/init.d/openvpn restart
+
+This will restart all configured OpenVPN tunnels.
+
+5. Verify the VPN is running
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+1. To ensure OpenVPN is using the correct configuration file and is running as expected, check the active processes: ::
+
+    ps xa | grep openvpn
+
+The output should resemble the following (example configuration name ``myvpn``): ::
+
+    4913 ?        S      0:00 /usr/sbin/openvpn --syslog openvpn(myvpn) --status /var/run/openvpn.myvpn.status --cd /etc/openvpn --config myvpn.ovpn --up /usr/libexec/openvpn-hotplug up myvpn --down /usr/libexec/openvpn-hotplug down myvpn --route-up /usr/libexec/openvpn-hotplug route-up myvpn --route-pre-down /usr/libexec/openvpn-hotplug route-pre-down myvpn --script-security 2
+
+Confirm the ``--config`` parameter points to the correct configuration file (e.g., ``myvpn.ovpn``).
+Ensure all references (e.g., ``myvpn``) match your intended VPN configuration.
+
+2. Check the OpenVPN logs to confirm the connection: ::
+
+    tail -f /var/log/messages | grep openvpn
+
+You should see log entries indicating a successful connection.
+
+
+.. note:: 
+
+  - **File Name Consistency:** The configuration name ``myvpn`` must match the OpenVPN section name in UCI and the configuration file's location. If you change the name, ensure all references to ``myvpn`` in commands and filenames are updated.
+  - **Automatic Startup:** By setting ``enabled='1'``, the VPN client will automatically start whenever the firewall boots.
+
+
+Optional: Disable Automatic Startup
```

Comment:
I'd move this section to the end.
```suggestion
Disable the tunnel
```

Reply:
> I'd move this section to the end.

Do you mean after the section: "Configure authentication credentials (optional)"
or aty the very end after the firewall section?

Reply:
I'd say to the very end

Reply:
> I'd say to the very end

Section moved